### PR TITLE
Added github action for testing container build on PRs

### PR DESCRIPTION
**Pull request by 0phelia36**
## TL;DR
Added a github action that tries to build the docker containers on all pull requests. Will let us automatically validate that PR's don't break the docker containers. Currently it is failing because #92 needed to add additional dependencies to the API container, the fix for that is in #94.

## What
What is affected by this PR?
- [ ] API
- [ ] GUI
- [ ] Hardware Integration
- [ ] OS/Deployment
- [ ] Documentation
- [X] Other (please describe in notes)

## Testing
How was this tested?
- [ ] Locally Virtualized
- [ ] Raspberry Pi
- [ ] With Hardware
- [X] Other (please describe in notes)

## Notes
Effects PRs.

### CODE REVIEW

**Review by 0x42red — APPROVED** [review]
LGTM assuming the billing for this is all handled and wont be a problem

**Comment by 0phelia36** [comment]
Github actions are free for public repo's so we should be able to have as many as we want
